src/components/document/document-workspace.tsx

```diff
@@ -207,6 +210,30 @@ export class DocumentWorkspaceComponent extends BaseComponent<IProps, {}> {
     }
   }
 
+  private handleDeleteDocument = (document: DocumentModelType) => {
+    this.stores.ui.confirm(`Delete this workspace?`, `Delete Workspace`)
+      .then((deleted: boolean) => {
+        const docType = document.type;
+        const invalidDocs = ProblemDocument || PublicationDocument || PersonalPublication ||
+                            LearningLogPublication || SupportPublication;
+        if (docType !== invalidDocs) {
+          document.setProperty("softDelete", "true");
+          this.handleDeleteOpenPrimaryDocument(deleted);
+        }
+      });
+  }
+
+  private handleDeleteOpenPrimaryDocument = async (softDelete: boolean) => {
```

Comment:
```suggestion
  private handleDeleteOpenPrimaryDocument = async () => {
```
This function should only be called if the delete has actually been confirmed by the user.